core/src/avm2/activation.rs

```diff
@@ -1203,22 +1306,23 @@ impl<'a, 'gc, 'gc_context> Activation<'a, 'gc, 'gc_context> {
     ) -> Result<FrameControl<'gc>, Error> {
         let multiname = self.pool_multiname(method, index)?;
         let object = self.context.avm2.pop().coerce_to_object(self)?;
-        let base_proto: Result<Object<'gc>, Error> = self
-            .base_proto()
-            .and_then(|p| p.proto())
-            .ok_or_else(|| "Attempted to get property on non-existent super object".into());
-        let base_proto = base_proto?;
-        let mut base = base_proto.construct(self, &[])?; //TODO: very hacky workaround
-
-        let name: Result<QName, Error> = base.resolve_multiname(&multiname)?.ok_or_else(|| {
-            format!(
-                "Could not resolve {:?} as super property",
-                multiname.local_name()
-            )
-            .into()
-        });
 
-        let value = base.get_property(object, &name?, self)?;
+        let name: Result<QName, Error> = object
+            .resolve_multiname(&multiname)?
+            .ok_or_else(|| format!("Could not find method {:?}", multiname.local_name()).into());
+        let name = name?;
+
+        let base_constr: Result<Object<'gc>, Error> = self
+            .base_constr()
+            .and_then(|bc| bc.base_class_constr())
+            .ok_or_else(|| {
+                "Attempted to call super method without a superclass."
+                    .to_string()
+                    .into()
+            });
+        let base_constr = base_constr?;
+
+        let value = base_constr.call_instance_getter(&name, Some(object), self)?;
```

Comment:
This is incorrectly assuming that the super property is a getter (ditto with `setsuper`). Consider:
```
class Base {
	var a: int = 1;
}
class Sub extends Base {
	function f() { return super.a; }
	function g() { super.a++; }
}
var sub = new Sub();
trace(sub.f());
sub.g();
trace(sub.f());
```